.buildkite/scripts/opentelemetry-benchmark.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+echo "--- Build opentelemetry-java-instrumentation"
+git clone https://github.com/open-telemetry/opentelemetry-java-instrumentation.git
+cd opentelemetry-java-instrumentation/
+./gradlew assemble
+
+echo "--- Download the latest elastic-agent.zip"
+cd benchmark-overhead
+ELASTIC_SNAPSHOT_URL=$(curl -s -u $GITHUB_USERNAME:$GITHUB_SECRET "https://api.github.com/repos/elastic/apm-agent-java/actions/workflows/49838992/runs?branch=main" | jq -c '.workflow_runs[] | {conclusion, updated_at, display_title, url}' | grep -v null  | grep -v pending | grep -v cancelled | grep success | head -1 | awk -F'":"' '{print $5}' | tr -d '"}')
+ELASTIC_SNAPSHOT_ARTIFACTS=$(curl -s -u $GITHUB_USERNAME:$GITHUB_SECRET "$ELASTIC_SNAPSHOT_URL" | grep artifacts_url | awk -F'":' '{print $2}' | tr -d '"} ,')
+ELASTIC_SNAPSHOT_ZIPFILE=$(curl -s -u $GITHUB_USERNAME:$GITHUB_SECRET "$ELASTIC_SNAPSHOT_ARTIFACTS" | jq -c ".artifacts[] | {name,archive_download_url}" | grep '"elastic-apm-agent"' | awk -F'":' '{print $3}' | tr -d '"}')
+curl -s --output "elastic-agent.zip" -L -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GITHUB_SECRET" -H "X-GitHub-Api-Version: 2022-11-28" -u $GITHUB_USERNAME:$GITHUB_SECRET "$ELASTIC_SNAPSHOT_ZIPFILE"
+unzip elastic-agent.zip
```

Comment:
This could be potentially simplify with the `gh cli` given the service account:
- https://cli.github.com/manual/gh_run_download

Reply:
refs https://github.com/elastic/apm-agent-java/pull/3371/commits/38b42a4e145f4b27d10307b2a97e2bd4a2c75d67